### ISSUE TITLE
Re-Introduce "Convert to Template Literal" for non-concatenated, valid string literals

### ISSUE THREAD

**Issue by automatensalat** [nicoespeon/abracadabra#94]
## Is this request related to a problem? Please describe.

With release 4.0.0 the `Convert to Template Literal` was removed from the extension. The explanation for this is that VSCode introduced a similar refactoring (see [here](https://devblogs.microsoft.com/typescript/announcing-typescript-3-8/#convert-to-template-string)). The change log suggests that the refactoring is now obsolete in Abracadabra, so it was removed. 

I am not so happy with this change, because the VSCode feature does not cover the same workflow that was possible with the Abracadabra refactoring. 
It seems VSCode works like this: you take an already written string concatination like 
```ts
const a = "I have " + numApples + " apples";
```
, apply the VSCode refactoring which changes the code to 
```ts 
const a = "I have ${numApples} apples";
```
This is useful, but it only works if you already have such a concatination. 

My workflow with the refactoring often looked like this: I have a string in which I now want to inject a dynamic value. So I could place the cursor into the string and apply the `Convert to Template Literal` refactoring, e.g.

```ts
const a = "I have some apples";
```
to
```ts 
const a = "I have ${numApples} apples";
```

This now longer works now.

## Describe the solution you'd like

Would you consider re-introducing the refactoring to the extension?

**Comment by nicoespeon** [comment]
Oh wow that's a good catch! I was looking for scenario VS Code native refactoring wouldn't handle and I forgot about this one.

Indeed, that's a good reason to bring back the refactoring.

However, I don't want to propose it when VS Code handles it (it would spam the user). So I guess we can bring it back and simplify it to only handle the case of string literals that are not concatenated 👍 

Also, I'll try to see if I can suggest this improvement directly on VS Code. The code for the refactoring is already implemented, it shouldn't  be too hard to propose an improvement to handle that kind of scenario natively.

**Comment by nicoespeon** [comment]
Here's the PR that implemented the refactoring in VS Code: https://github.com/microsoft/TypeScript/pull/30565 (it's a Code Action provided by TypeScript).

**Comment by nicoespeon** [comment]
Actually, this is an intended feature of the Typescript refactoring: https://github.com/microsoft/TypeScript/issues/36784

So yeah, we'll put back Abracadabra refactoring on string literals while this is being discussed. It might take some time, if it's ever got accepted.

**Comment by nicoespeon** [comment]
There you go!

![convert-to-template-literal](https://user-images.githubusercontent.com/1094774/78993564-eed4c680-7b0b-11ea-8b64-1a8049373c3c.gif)

**Comment by nicoespeon** [comment]
Released in v4.1.0
